util/util.go

```diff
@@ -1,15 +1,88 @@
 package util
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"os"
 	"strings"
+	"time"
 	"unicode"
 	"unicode/utf8"
 
 	"github.com/algorand/go-codec/codec"
+
+	log "github.com/sirupsen/logrus"
 )
 
+// ErrTimeout is returned when CallWithTimeout has a normal timeout.
+var ErrTimeout = errors.New("timeout during call")
+
+// IsTimeoutError compares the given error against the timeout errors.
+func IsTimeoutError(err error) bool {
+	if err == nil {
```

Comment:
I think this one is OK. The return value from `CallWithTimeout` happens to be `ErrTimeout`, but it could just as easily have been some context deadline error. By having a function the implementation detail isn't exposed outside of the util file.

Reply:
I see. But `if err == nil {` is still unnecessary.